library/src/handle.cpp

```diff
@@ -1,28 +1,78 @@
-
+/* ************************************************************************
+ * Copyright 2016 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
 #include "definitions.h"
 #include "status.h"
 #include "handle.h"
 #include <hip/hip_runtime_api.h>
+#include <unistd.h>
+#include <pwd.h>
 
 /*******************************************************************************
  * constructor
  ******************************************************************************/
-_rocblas_handle::_rocblas_handle()
+_rocblas_handle::_rocblas_handle() : layer_mode(rocblas_layer_mode_logging)
 {
-
     // default device is active device
     THROW_IF_HIP_ERROR(hipGetDevice(&device));
     THROW_IF_HIP_ERROR(hipGetDeviceProperties(&device_properties, device));
 
     // rocblas by default take the system default stream 0 users cannot create
+
+    // set layer_mode from vaule of environment variable ROCBLAS_LAYER
+    char* str_layer_mode;
+    if((str_layer_mode = getenv("ROCBLAS_LAYER")) == NULL)
+    {
+        layer_mode = rocblas_layer_mode_none;
+    }
+    else
+    {
+        int int_layer_mode = atoi(str_layer_mode);
+
+        layer_mode = (rocblas_layer_mode)int_layer_mode;
+    }
+
+    // open log file
+    if(layer_mode & rocblas_layer_mode_logging)
+    {
+        // construct filepath for log file in home directory
+        const char* homedir_char;
+        std::string homedir_str;
+
+        if((homedir_char = getenv("HOME")) == NULL)
```

Comment:
Instead of reading the HOME env variable, we should attempt to write log files into the PWD.  First, you have to check if the PWD is writeable, and if it is not, then the user should be able to specify an alternate logging directory with an overriding ENV, ROCBLAS_LOG_DIR or some such.

Reply:
fixed: log file will be in current working directory if it is writable, else in ROCBLAS_LOG_DIR